src/Shared/FileUtilities.cs

```diff
@@ -421,6 +421,72 @@ internal static string FixFilePath(string path)
         }
 
         /// <summary>
+        /// If on Unix, convert backslashes to slashes for strings that resemble paths.
+        /// The heuristic is if something resembles paths (contains slashes) check if the
+        /// first segment exists and is a directory.
+        /// Use a native shared method to massage file path. If the file is adjusted,
+        /// that qualifies is as a path.
+        /// </summary>
+        internal static string MaybeAdjustFilePath(string value)
+        {
+            if (NativeMethodsShared.IsWindows || string.IsNullOrWhiteSpace(value) ||
+                value.StartsWith("$(") || value.StartsWith("@(") || value.StartsWith("\\\\"))
+            {
+                return value;
```

Comment:
I don't know what one can expect to see as far as number of properties going through here. I assume 1000 isn't abnormal? 10,000?